multisrc/src/main/java/eu/kanade/tachiyomi/multisrc/colamanga/ColaManga.kt

```diff
@@ -0,0 +1,336 @@
+
+import android.annotation.SuppressLint
+import android.app.Application
+import android.os.Handler
+import android.os.Looper
+import android.view.View
+import android.webkit.JavascriptInterface
+import android.webkit.WebView
+import androidx.preference.ListPreference
+import androidx.preference.PreferenceScreen
+import eu.kanade.tachiyomi.lib.synchrony.Deobfuscator
+import eu.kanade.tachiyomi.network.GET
+import eu.kanade.tachiyomi.network.interceptor.rateLimitHost
+import eu.kanade.tachiyomi.source.ConfigurableSource
+import eu.kanade.tachiyomi.source.model.FilterList
+import eu.kanade.tachiyomi.source.model.MangasPage
+import eu.kanade.tachiyomi.source.model.Page
+import eu.kanade.tachiyomi.source.model.SChapter
+import eu.kanade.tachiyomi.source.model.SManga
+import eu.kanade.tachiyomi.source.online.ParsedHttpSource
+import kotlinx.serialization.json.Json
+import kotlinx.serialization.json.jsonArray
+import kotlinx.serialization.json.jsonObject
+import kotlinx.serialization.json.jsonPrimitive
+import okhttp3.HttpUrl.Companion.toHttpUrl
+import okhttp3.Request
+import org.jsoup.nodes.Document
+import org.jsoup.nodes.Element
+import rx.Observable
+import uy.kohesive.injekt.Injekt
+import uy.kohesive.injekt.api.get
+import uy.kohesive.injekt.injectLazy
+import java.util.concurrent.CountDownLatch
+import java.util.concurrent.TimeUnit
+
+abstract class ColaManga(
+    final override val name: String,
+    final override val baseUrl: String,
+    final override val lang: String,
+) : ParsedHttpSource(), ConfigurableSource {
+
+    override val supportsLatest = true
+
+    private val json: Json by injectLazy()
+
+    private val intl = ColaMangaIntl(lang)
+
+    private val preferences by lazy {
+        Injekt.get<Application>().getSharedPreferences("source_$id", 0x0000)
+    }
+
+    override val client = network.cloudflareClient.newBuilder()
+        .rateLimitHost(
+            baseUrl.toHttpUrl(),
+            preferences.getString(RATE_LIMIT_PREF_KEY, RATE_LIMIT_PREF_DEFAULT)!!.toInt(),
+            preferences.getString(RATE_LIMIT_PERIOD_PREF_KEY, RATE_LIMIT_PERIOD_PREF_DEFAULT)!!.toLong(),
+            TimeUnit.MILLISECONDS,
+        )
+        .addInterceptor(ColaMangaImageInterceptor())
+        .build()
+
+    override fun headersBuilder() = super.headersBuilder()
+        .add("Origin", baseUrl)
+        .add("Referer", "$baseUrl/")
+
+    override fun popularMangaRequest(page: Int) =
+        GET("$baseUrl/show?orderBy=dailyCount&page=$page", headers)
+
+    override fun popularMangaSelector() = "li.fed-list-item"
+
+    override fun popularMangaFromElement(element: Element) = SManga.create().apply {
+        element.selectFirst("a.fed-list-title")!!.let {
+            setUrlWithoutDomain(it.attr("href"))
+            title = it.text()
+        }
+        thumbnail_url = element.selectFirst("a.fed-list-pics")?.absUrl("data-original")
+    }
+
+    override fun popularMangaNextPageSelector() = "a:contains(下页):not(.fed-btns-disad)"
```

Comment:
this selector works on the english version?

Reply:
fixed